functions.py

```diff
@@ -26,12 +34,223 @@ def getServiceFromUser():
     service= getUserInput("What service do you want to use?\n");
     return service
 
-def writeToFile(service, pw):
-    fileName = "./passwords.txt"
-    passwordFile = open(fileName, 'a')
+def repromptLogin():
+        print("That doesn't seem to match any of our records...\n")
+
+        print("Try again or go back to menu?\n")
+        print("(1) Try Again")
+        print("(2) Main Menu")
+        choice = getUserInput("> ")
+        if choice == "1":
+            loginUser()
+        elif choice == "2":
+            handleLogin()
+        else:
+            print("I didn't recognize that input")
+            repromptLogin()
+
+def loginUser():
+    username = getUserInput("Please enter your username\n> ")
+    pw = getUserInput("Please enter your password\n> ", True)
+    pw = hashlib.sha512(pw.encode('utf-8')).hexdigest()
+    if checkUserCredentials(username, pw):
+        return True
+    else:
+        repromptLogin()
+
+
+def signUpUser():
+    username = getUserInput("Please enter your username\n> ")
+    pw = getUserInput("Please enter your password\n> ", True)
+    pw = hashlib.sha512(pw.encode('utf-8')).hexdigest()
+    if addUser(username, pw):
+        return True
+    else:
+        print("Sorry, that username is already taken")
+        signUpUser()
+
+def handleLogin():
+    print("Do you want to log in or start a new account?\n(Enter the number of your choice)\n")
+    print("(1) Log In")
+    print("(2) Start New Account\n")
+    option = getUserInput("> ")
+    if option == "1":
+        loginUser()
+
+    elif option == "2":
+        signUpUser()
+
+    else:
+        print("Please enter a valid option\n")
+        handleLogin()
+
+def welcomeMessage():
+    print("\n\n")
+    print("##################################################")
+    print("# Welcome to Passman!")
+    print("##################################################\n\n")
+
+def generatePasswordPrompt():
+    #TODO
+    print("todo")
+
+def listServicesPrompt():
+    for service in getAllServices():
```

Comment:
@nosnixster Since you're just printing stuff out here, you can leave the variables as empty strings and print regardless. It's better to keep a static format anyway, IMO.

I rewrote this to be more straight forward here: 0ce202f